ref/go/src/bech32/bech32_test.go

```diff
@@ -0,0 +1,242 @@
+		}
+	}
+}
+
+// add coverage tests
+
+func TestCoverage(t *testing.T) {
+	var err error
+	_, err = bech32.SegwitAddrEncode("bc", 0, []int{-1})
+	if err == nil {
+		t.Errorf("Coverage SegwitAddrEncode invalid data range error case : FAIL")
+	} else {
+		t.Log("Coverage SegwitAddrEncode invalid data range error case : ok / error :", err)
+	}
+	_, err = bech32.SegwitAddrEncode("bc", 33, []int{})
+	if err == nil {
+		t.Errorf("Coverage SegwitAddrEncode invalid data error case : FAIL")
+	} else {
+		t.Log("Coverage SegwitAddrEncode invalid data error case : ok / error :", err)
+	}
+	_, err = bech32.SegwitAddrEncode("bc", 16, []int{0, 1, 2, 3, 4, 5})
+	if err != nil {
+		t.Errorf("Coverage SegwitAddrEncode witness version normal case : FAIL / error : %+v\n", err)
+	} else {
+		t.Log("Coverage SegwitAddrEncode witness version normal case : ok")
+	}
+	_, err = bech32.SegwitAddrEncode("bc", 17, []int{0, 1, 2, 3, 4, 5})
+	if err == nil {
+		t.Errorf("Coverage SegwitAddrEncode invalid witness version error case : FAIL")
+	} else {
+		t.Log("Coverage SegwitAddrEncode invalid witness version error case : ok / error :", err)
+	}
+	_, _, err = bech32.SegwitAddrDecode("a", "A12UEL5L")
+	if err == nil {
+		t.Errorf("Coverage SegwitAddrDecode invalid decode data length error case : FAIL")
+	} else {
+		t.Log("Coverage SegwitAddrDecode invalid decode data length error case : ok / error :", err)
+	}
+	_, _, err = bech32.Decode("a1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq")
+	if err == nil {
+		t.Errorf("Coverage Decode too long error case : FAIL")
+	} else {
+		t.Log("Coverage Decode too long error case : ok / error :", err)
+	}
+	_, _, err = bech32.Decode("1")
+	if err == nil {
+		t.Errorf("Coverage Decode separator '1' at invalid position error case : FAIL")
+	} else {
+		t.Log("Coverage Decode separator '1' at invalid position error case : ok / error :", err)
+	}
+	_, _, err = bech32.Decode("a1qqqqq")
+	if err == nil {
+		t.Errorf("Coverage Decode separator '1' at invalid position error case : FAIL")
+	} else {
+		t.Log("Coverage Decode separator '1' at invalid position error case : ok / error :", err)
+	}
+	_, _, err = bech32.Decode("a" + string(32) + "1qqqqqq")
+	if err == nil {
+		t.Errorf("Coverage Decode invalid character human-readable part error case : FAIL")
+	} else {
+		t.Log("Coverage Decode invalid character human-readable part error case : ok / error :", err)
+	}
+	_, _, err = bech32.Decode("a" + string(127) + "1qqqqqq")
+	if err == nil {
+		t.Errorf("Coverage Decode invalid character human-readable part error case : FAIL")
+	} else {
+		t.Log("Coverage Decode invalid character human-readable part error case : ok / error :", err)
+	}
+	_, _, err = bech32.Decode("a1qqqqqb")
+	if err == nil {
+		t.Errorf("Coverage Decode Invalid program length error case : FAIL")
+	} else {
+		t.Log("Coverage Decode Invalid program length case : ok / error :", err)
+	}
+
+	data := make([]int, 32)
```

Comment:
Remove this section? The Decode() fails because of the empty human readable part, so the unit test fails.